p2p/simulations/network.go

```diff
@@ -168,7 +168,6 @@ func (net *Network) Start(id enode.ID) error {
 // snapshots
 func (net *Network) startWithSnapshots(id enode.ID, snapshots map[string][]byte) error {
 	net.lock.Lock()
-	defer net.lock.Unlock()
 
 	node := net.getNode(id)
 	if node == nil {
```

Comment:
I think if we get an error here, we will never unlock the mutex. This is true for all the error cases in this function.

Reply:
good catch. fixed all of the corresponding error handlers